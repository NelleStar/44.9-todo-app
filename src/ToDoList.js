import { useState } from "react";
import { v4 as uuid } from "uuid";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const ToDoList = () => {
    const INITIAL_STATE = []
    const [todos, setTodos] = useState(INITIAL_STATE)

    const addTodo = ( newTodo ) => {
        setTodos(todos => [...todos, { ...newTodo, id: uuid() }])
    }

    const handleDelete = (id) => {
        const updatedTodos = todos.filter(todo => todo.id !== id);
        setTodos(updatedTodos)
    }

    return(
        <>
            <h1>ToDo List</h1>
            <NewTodoForm addTodo={addTodo} />
            <div>
                {todos.map(({ id, todo }) => 
                    <Todo 
                        key={id} 
                        id={id} 
                        todo={todo} 
                        onDelete={handleDelete}
                    />
                )}
            </div>
        </>
    )
}

export default ToDoList;