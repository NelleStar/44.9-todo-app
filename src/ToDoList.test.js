import { render, fireEvent } from "@testing-library/react";
import ToDoList from "./ToDoList";

// smoke test
it("renders without crashing", function () {
  render(<ToDoList />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<ToDoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new item", function () {
  const { queryByText, getByLabelText } = render(<ToDoList />);
  const input = getByLabelText("ToDo:");
  const btn = queryByText("Add Item");
  expect(queryByText("Walk Dog")).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "Walk Dog" } });
  fireEvent.click(btn);
  expect(queryByText("Walk Dog")).toBeInTheDocument();
});
