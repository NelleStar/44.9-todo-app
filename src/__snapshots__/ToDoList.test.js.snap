// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    ToDo List
  </h1>
  <form>
    <label
      for="todo"
    >
      ToDo: 
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button>
      Add Item
    </button>
  </form>
  <div />
</DocumentFragment>
`;
